cache_archive.go

```diff
@@ -53,8 +53,8 @@ func NewArchive(pth string, compress bool) (*Archive, error) {
 }
 
 // Write writes the given files in the cache archive.
-func (a *Archive) Write(pths []string) error {
-	for _, pth := range pths {
+func (a *Archive) Write(pathToIndicator map[string]string) error {
```

Comment:
why did we change this function's interface? it is still using only the path

Reply:
Seemed unnecessary to copy over the paths, can not see performance difference tough.